command/agent/config.go

```diff
@@ -328,6 +328,27 @@ type ServerConfig struct {
 	EncryptKey string `mapstructure:"encrypt" json:"-"`
 }
 
+// UpdateTLSConfig will reload an agent's TLS configuration. If there is an error
+// while loading key and certificate files, the agent will remain at its
+// current configuration and return an error.
+// This only allows reloading the certificate and keyfile- other TLSConfig
+// fields are ignored.
+func (c *Config) UpdateTLSConfig(newConfig *config.TLSConfig) error {
```

Comment:
Does this method need locking? If readers are accessing `c.TLSConfig.CertFile` concurrently with `Reload()`s then there's a race.

Reply:
Good point, added.